File: app/src/main/java/com/example/olesya/rxjavatest/BoundService.java
package com.example.olesya.rxjavatest;

import android.app.Service;
import android.arch.lifecycle.MutableLiveData;
import android.content.Intent;
import android.os.Binder;
import android.os.IBinder;
import android.support.annotation.Nullable;

//TODO: может это сделать не интерфейсом а классом, который наследует сервис и посмотреть чо общего в общении клиента и сервера
//TODO: потому что там они общаются одинаково плюс минус
public class BoundService extends Service {

    protected final int PORT_NUMBER = 8888;
    protected MutableLiveData<String> message = new MutableLiveData<>();
    protected MyBinder binder = new BoundService.MyBinder();

    public void setMessage(MutableLiveData<String> message) {
        this.message = message;
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    public class MyBinder extends Binder {
        BoundService getService() {
            return BoundService.this;
        }
    }
}

File: app/src/main/java/com/example/olesya/boardgames/connection/common/BoundService.kt
package com.example.olesya.boardgames.connection.common

import android.app.Service
import android.arch.lifecycle.MutableLiveData
import android.content.Intent
import android.os.Binder
import android.os.IBinder

open class BoundService: Service() {

    val PORT_NUMBER = 8888

    var screenMessage: MutableLiveData<String> = MutableLiveData()

    val binder: MyBinder = MyBinder()

    override fun onBind(intent: Intent?): IBinder? {
        return binder
    }

    inner class MyBinder: Binder() {
        fun getService(): BoundService {
            return this@BoundService
        }
    }

}